conda_recipe_manager/commands/graph.py

```diff
@@ -34,8 +35,7 @@ def graph(path: Path) -> None:
     recipe_graph: Final[RecipeGraph] = RecipeGraphFromDisk(path)
     if not recipe_graph:
         print_err(f"The path provided does not contain any recipe files: {path}")
-        # TODO use enum scheme
-        sys.exit(1)
+        sys.exit(ExitCode.CLICK_USAGE)
```

Comment:
Any reason to not use click's native abort system? Something like `ctx.fail(f"The path provided does not contain any recipe files: {path}")`

Reply:
It doesn't appear that that function allows you to control the error codes. I have have this standardized enumeration to ensure that all the CRM commands emit similar error codes and that I control what the error codes are.

Trying to go for that UNIX design philosophy and allow other programs to rely on error codes having a particular meaning.

Reply:
Right, that's a good policy, I like it! For custom exit codes click has built-in error handling via Python exceptions: https://click.palletsprojects.com/en/8.1.x/exceptions/#where-are-errors-handled

Here's the code that does it: https://github.com/pallets/click/blob/99015e1456b6952054fda8312b40ea646737168d/src/click/core.py#L1320-L1324. 

[`ClickException` has a bunch of subclasses that you could reuse, if they match your use case](https://github.com/pallets/click/blob/99015e1456b6952054fda8312b40ea646737168d/src/click/exceptions.py#L25). The key part is that for custom exit codes, simply subclass and define an `exit_code` class attribute with your values.